viewer/packages/tools/src/Measurement/MeasurementManager.ts

```diff
@@ -0,0 +1,183 @@
+/*!
+ * Copyright 2022 Cognite AS
+ */
+
+import assert from 'assert';
+import * as THREE from 'three';
+import { HtmlOverlayTool } from '../HtmlOverlay/HtmlOverlayTool';
+import { MeasurementLabels } from './MeasurementLabels';
+import { MeasurementLine } from './MeasurementLine';
+import { MeasurementOptions } from './types';
+
+export type Measurement = {
+  readonly measurementId: number;
+  readonly startPoint: THREE.Vector3;
+  readonly endPoint: THREE.Vector3;
+  readonly distanceInMeters: number;
+};
+
+export class MeasurementManager {
+  private readonly _measurementLabel: MeasurementLabels;
+  private _line: MeasurementLine | null = null;
```

Comment:
Dont have this has null - the only reason you have it as nullable is that `startMeasurement` initializes it. Instead, have `startMeasurement` return `Measurement` and assign in constructor and add to `meshGroup`.

Avoid `nullable` whenever possible.